templates/base.html

```diff
@@ -29,12 +29,35 @@
       gtag('config', 'G-D66RRD8GF2');
     </script>
 
+    <link rel="prefetch" href="{{ get_versioned_filename('/static/js/highstock.js') }}">
+    <link rel="prefetch" href="{{ get_versioned_filename('/static/js/highcharts-more.js') }}">
+    <link rel="prefetch" href="{{ get_versioned_filename('/static/js/report.js') }}">
+    <link rel="prefetch" href="{{ get_versioned_filename('/static/js/timeseries.js') }}">
   </head>
   <body>
     {% block content %}{% endblock %}
     {% block scripts %}
     <script defer src="{{ get_versioned_filename('/static/js/web-vitals.js') }}" nonce="{{ csp_nonce() }}"></script>
     <script defer src="{{ get_versioned_filename('/static/js/send-web-vitals.js') }}" nonce="{{ csp_nonce() }}"></script>
+    <script type="speculationrules" nonce="{{ csp_nonce() }}">
+    {
+      "prefetch": [{
+        "source": "document",
```

Comment:
Technically this is not needed as the `where` clause implies this, and it may be removed in future. There was one version of Chrome (121) where document rules was supported but needs this explicitly set, before we made it optional (in Chrome 122):

https://developer.chrome.com/blog/speculation-rules-improvements?hl=en#optional_source

```suggestion
```